src/app/app.component.ts

```diff
@@ -132,6 +133,68 @@ export class AppComponent implements OnInit {
 
   @ViewChild('uploader') uploader: FileUploadComponent;
 
+  pillboxData = {
```

Comment:
Expanding on what I mentioned above, maybe you need a PillboxContentType type?
`type PillboxContentType = PillboxContent[] | PillboxContentPair | PillboxFraction `

If you had a type, then you can type pillboxData as a key/value pair of PillboxContentType objects and not have to specify <PillboxContent[]> each time (I think...)

It can be simplified to something like:

```
pillboxData: {[key: string]: PillboxContentType} = {
  // PillboxContent[]
  simplest: [{
    label: 'Single label, no styling or whatever'
  }],
  
  // PillboxContentPair
  twoLabels: {
    first: [{
        label: 'First label',
      }],
    second: [{
        label: 'Second label',
    }],
  },
      
  // PillboxFraction 
  fraction: {
    numerator: 95,
    denominator: 96,
    rate: 0.95,
    rateTooltip: 'Higher rates indicate better performance'
  }
}
```

NOTE: Syntax may not be correct, just providing a rough untested example...

Reply:
Hmmm... yeah I'll try this first before trying to recreate the error I described with the multiple allowed `@Input` types.